text/0517-io-os-reform.md

```diff
@@ -483,7 +483,72 @@ throughout IO, we can go on to explore the modules in detail.
 ### `std::env`
 [std::env]: #stdenv
 
-> To be added in a follow-up PR.
+Most of what's available in `std::os` today will move to `std::env`,
+and the signatures will be updated to follow this RFC's
+[Design principles] as follows.
+
+**Arguments**:
+
+* `args`: change to yield an iterator rather than vector if possible; in any case, it should produce an `OsStrBuf`.
```

Comment:
>On linux (and freebsd I believe) we currently don't have implemented an easy way to fetch argc/argv from the system, so we squirrel them away ourselves.

Why do you need to fetch arguments? The arguments are provided to the runtime or there are none. That's how it works right now and nobody seems to be complaining about it.